libobs/obs.h

```diff
@@ -2457,6 +2457,30 @@ EXPORT void obs_encoder_set_gpu_scale_type(obs_encoder_t *encoder,
 EXPORT bool obs_encoder_set_frame_rate_divisor(obs_encoder_t *encoder,
 					       uint32_t divisor);
 
+/**
+ * Adds region of interest (ROI) for an encoder. This allows prioritizing
+ * quality of regions of the frame.
+ * If regions overlap, regions added earlier take precedence.
+ *
+ * Returns false if the encoder does not support ROI or region is invalid.
+ */
+EXPORT bool obs_encoder_add_roi(obs_encoder_t *encoder,
+				const struct region_of_interest *roi);
+/** For video encoders, returns true if any ROIs were set */
+EXPORT bool obs_encoder_has_roi(const obs_encoder_t *encoder);
+/** Remove specified region from encoder */
+EXPORT void obs_encoder_remove_roi(obs_encoder_t *encoder,
+				   const struct region_of_interest *roi);
+/** Clear all regions */
+EXPORT void obs_encoder_clear_roi(obs_encoder_t *encoder);
+/** Enumerate regions with callback (reverse order of addition) */
+EXPORT void obs_encoder_enum_roi(obs_encoder_t *encoder,
+				 void (*enum_proc)(void *,
+						   struct region_of_interest *),
+				 void *param);
+/** Get ROI increment, encoders must rebuild their ROI map if it has changed */
+EXPORT uint32_t obs_encoder_get_roi_increment(const obs_encoder_t *encoder);
```

Comment:
For some reason using a uint32_t for this feels risky. Maybe just ensure that nothing breaks if this value wraps and add a note to the docs?

Reply:
I don't think anybody is going to update the ROIs enough to exceed 4.2 billion revisions in one OBS runtime (if you make one change every frame that would take over 2 years). Even if it rolls over, since the comparison should be `!=` not `>`, it should be fine.